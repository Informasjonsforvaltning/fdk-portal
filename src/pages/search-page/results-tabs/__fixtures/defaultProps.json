{
  "searchParam": "",
  "countDatasets": 985,
  "countTerms": 1,
  "countApis": 4
}
