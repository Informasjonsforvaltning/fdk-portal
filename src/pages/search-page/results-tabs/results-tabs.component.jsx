import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import cx from 'classnames';
import { withRouter } from 'react-router';

import localization from '../../../lib/localization';
import './results-tabs.scss';
import {
  PATHNAME_DATASETS,
  PATHNAME_APIS,
  PATHNAME_CONCEPTS,
  PATHNAME_INFORMATIONMODELS
} from '../../../constants/constants';
import { getLinkForTab } from '../search-location-helper';

export const ResultsTabsPure = ({
  countDatasets,
  countTerms,
  countApis,
  countInformationModels,
  location
}) => {
  const activePath = location.pathname;
  return (
    <ul className="search-results-tabs d-flex align-items-center justify-content-center flex-wrap">
      <li
        className={cx('d-flex justify-content-center align-self-center', {
          'li-active': activePath === PATHNAME_DATASETS
        })}
      >
        <Link
          className="d-flex justify-content-center"
          to={getLinkForTab(location, PATHNAME_DATASETS)}
          aria-label="Link til side for datasett:"
        >
          <i className="mr-2 search-results-tab-dataset" />

          <div className="align-self-center">
            {localization.page.datasetTab}&nbsp;({countDatasets})
          </div>
        </Link>
      </li>
      <li
        className={cx('d-flex justify-content-center', {
          'li-active': activePath === PATHNAME_APIS
        })}
      >
        <Link
          className="d-flex justify-content-center"
          to={getLinkForTab(location, PATHNAME_APIS)}
          aria-label="Link til side for api:"
        >
          <i className="mr-2 search-results-tab-api" />

          <div className="align-self-center">
            {localization.page.apiTab}&nbsp;({countApis})
          </div>
        </Link>
      </li>
      <li
        className={cx('d-flex justify-content-center', {
          'li-active': activePath === PATHNAME_CONCEPTS
        })}
      >
        <Link
          className="d-flex justify-content-center"
          to={getLinkForTab(location, PATHNAME_CONCEPTS)}
          aria-label="Link til side for begrep:"
        >
          <i className="mr-2 search-results-tab-term" />

          <div className="align-self-center">
            {localization.page.termTab}&nbsp;({countTerms})
          </div>
        </Link>
      </li>
      <li
        className={cx('d-flex justify-content-center beta', {
          'li-active': activePath === PATHNAME_INFORMATIONMODELS
        })}
      >
        <Link
          className="d-flex justify-content-center"
          to={getLinkForTab(location, PATHNAME_INFORMATIONMODELS)}
          aria-label="Link til side for informasjonsmodell:"
        >
          <i className="mr-2 search-results-tab-infomodel" />

          <div className="align-self-center mr-4">
            {localization.page.informationModelTab}&nbsp;(
            {countInformationModels})
          </div>
        </Link>
      </li>
    </ul>
  );
};

ResultsTabsPure.defaultProps = {
  countDatasets: null,
  countTerms: null,
  countApis: null,
  countInformationModels: null,

  location: { search: '' }
};

ResultsTabsPure.propTypes = {
  countDatasets: PropTypes.number,
  countTerms: PropTypes.number,
  countApis: PropTypes.number,
  countInformationModels: PropTypes.number,

  location: PropTypes.object
};

export const ResultsTabs = withRouter(ResultsTabsPure);
