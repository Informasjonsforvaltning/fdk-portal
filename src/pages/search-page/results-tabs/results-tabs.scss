@import '../../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.search-results-tabs {
  margin-bottom: 0;
  padding-left: 0;
  li {
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
    border-color: $color-danger;
    list-style-type: none;
    button, a {
      background: none;
      border: 0;
      box-sizing: border-box;
      color: $color-link;
      padding: .4em 1em;
      font-weight: 500;
      text-decoration: none;
    }
  }

  .li-active {
    color: $color-link;
    background-color: $color-neutral-lightest;
    font-size: 1.8rem;

    &.beta {
      background-image: url("/img/banner-beta.svg");
      background-repeat: no-repeat;
      background-position: right;
      background-size: 2.4em;
    }

    button, a {
      color: $color-neutral-darkest !important;
    }
  }
}

/* Medium Devices, Desktops */
@media only screen and (min-width : 992px) {

  .search-results-tabs {

    li {
      line-height: 1em;
      min-width: 10em;
      button, a {
        padding: 0.6em 0.8em;
      }
    }

    .li-active {
      font-size: 1.8rem;

      &.beta {
        background-image: url("/img/banner-beta.svg");
        background-repeat: no-repeat;
        background-position: right;
        background-size: 2.7em;
      }

      
    }
  }
}
.search-results-tab-dataset {
  background-image: url("/img/icon-catalog-dataset-md.svg");
  width: 1.5em; 
  height: 1.5em;
  display: inline-block;
}

.search-results-tab-api {
  background-image: url("/img/icon-catalog-api-md.svg");
  width: 1.5em; 
  height: 1.5em;
  display: inline-block;
}

.search-results-tab-term {
  background-image: url("/img/icon-catalog-concept-md.svg");
  width: 1.5em; 
  height: 1.5em;
  display: inline-block;
}

.search-results-tab-infomodel {
  background-image: url("/img/icon-catalog-infomodel-md.svg");
  width: 1.5em; 
  height: 1.5em;
  display: inline-block;
}
