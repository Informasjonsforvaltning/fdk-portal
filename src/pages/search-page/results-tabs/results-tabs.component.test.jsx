import React from 'react';
import { shallow } from 'enzyme';
import { ResultsTabsPure } from './results-tabs.component';
import {
  PATHNAME_DATASETS,
  PATHNAME_CONCEPTS
} from '../../../constants/constants';
import defaultProps from './__fixtures/defaultProps.json';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<ResultsTabsPure {...defaultProps} />);
});

test('should render ResultsTabs with "Dataset" active correctly', () => {
  wrapper.setProps({
    location: { pathname: PATHNAME_DATASETS, search: '' }
  });
  expect(wrapper).toMatchSnapshot();
});

test('should render ResultsTabs with "Concepts" active correctly', () => {
  wrapper.setProps({
    location: { pathname: PATHNAME_CONCEPTS, search: '' }
  });
  expect(wrapper).toMatchSnapshot();
});
