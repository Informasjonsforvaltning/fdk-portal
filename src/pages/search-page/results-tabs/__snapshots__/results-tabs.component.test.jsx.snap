// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ResultsTabs with "Concepts" active correctly 1`] = `
<ul
  className="search-results-tabs d-flex align-items-center justify-content-center flex-wrap"
>
  <li
    className="d-flex justify-content-center align-self-center"
  >
    <Link
      aria-label="Link til side for datasett:"
      className="d-flex justify-content-center"
      replace={false}
      to={
        Object {
          "pathname": "/",
          "search": "",
        }
      }
    >
      <i
        className="mr-2 search-results-tab-dataset"
      />
      <div
        className="align-self-center"
      >
        Datasets
         (
        985
        )
      </div>
    </Link>
  </li>
  <li
    className="d-flex justify-content-center"
  >
    <Link
      aria-label="Link til side for api:"
      className="d-flex justify-content-center"
      replace={false}
      to={
        Object {
          "pathname": "/apis",
          "search": "",
        }
      }
    >
      <i
        className="mr-2 search-results-tab-api"
      />
      <div
        className="align-self-center"
      >
        APIs
         (
        4
        )
      </div>
    </Link>
  </li>
  <li
    className="d-flex justify-content-center li-active"
  >
    <Link
      aria-label="Link til side for begrep:"
      className="d-flex justify-content-center"
      replace={false}
      to={
        Object {
          "pathname": "/concepts",
          "search": "",
        }
      }
    >
      <i
        className="mr-2 search-results-tab-term"
      />
      <div
        className="align-self-center"
      >
        Concepts
         (
        1
        )
      </div>
    </Link>
  </li>
  <li
    className="d-flex justify-content-center beta"
  >
    <Link
      aria-label="Link til side for informasjonsmodell:"
      className="d-flex justify-content-center"
      replace={false}
      to={
        Object {
          "pathname": "/informationmodels",
          "search": "",
        }
      }
    >
      <i
        className="mr-2 search-results-tab-infomodel"
      />
      <div
        className="align-self-center mr-4"
      >
        Information models
         (
        )
      </div>
    </Link>
  </li>
</ul>
`;

exports[`should render ResultsTabs with "Dataset" active correctly 1`] = `
<ul
  className="search-results-tabs d-flex align-items-center justify-content-center flex-wrap"
>
  <li
    className="d-flex justify-content-center align-self-center li-active"
  >
    <Link
      aria-label="Link til side for datasett:"
      className="d-flex justify-content-center"
      replace={false}
      to={
        Object {
          "pathname": "/",
          "search": "",
        }
      }
    >
      <i
        className="mr-2 search-results-tab-dataset"
      />
      <div
        className="align-self-center"
      >
        Datasets
         (
        985
        )
      </div>
    </Link>
  </li>
  <li
    className="d-flex justify-content-center"
  >
    <Link
      aria-label="Link til side for api:"
      className="d-flex justify-content-center"
      replace={false}
      to={
        Object {
          "pathname": "/apis",
          "search": "",
        }
      }
    >
      <i
        className="mr-2 search-results-tab-api"
      />
      <div
        className="align-self-center"
      >
        APIs
         (
        4
        )
      </div>
    </Link>
  </li>
  <li
    className="d-flex justify-content-center"
  >
    <Link
      aria-label="Link til side for begrep:"
      className="d-flex justify-content-center"
      replace={false}
      to={
        Object {
          "pathname": "/concepts",
          "search": "",
        }
      }
    >
      <i
        className="mr-2 search-results-tab-term"
      />
      <div
        className="align-self-center"
      >
        Concepts
         (
        1
        )
      </div>
    </Link>
  </li>
  <li
    className="d-flex justify-content-center beta"
  >
    <Link
      aria-label="Link til side for informasjonsmodell:"
      className="d-flex justify-content-center"
      replace={false}
      to={
        Object {
          "pathname": "/informationmodels",
          "search": "",
        }
      }
    >
      <i
        className="mr-2 search-results-tab-infomodel"
      />
      <div
        className="align-self-center mr-4"
      >
        Information models
         (
        )
      </div>
    </Link>
  </li>
</ul>
`;
